homeassistant/components/triggercmd/config_flow.py

```diff
@@ -0,0 +1,72 @@
+"""Config flow for TRIGGERcmd integration."""
+
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+import jwt
+from triggercmd import client
+import voluptuous as vol
+
+from homeassistant import config_entries, exceptions
+from homeassistant.config_entries import ConfigFlowResult
+from homeassistant.core import HomeAssistant
+
+from .const import DOMAIN, TOKEN
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema({(TOKEN): str})
+
+
+async def validate_input(hass: HomeAssistant, data: dict) -> dict[str, Any]:
+    """Validate the user input allows us to connect.
+
+    Data has the keys from DATA_SCHEMA with values provided by the user.
+    """
+    if len(data[TOKEN]) < 100:
+        raise InvalidToken
+
+    tokenData = jwt.decode(data[TOKEN], options={"verify_signature": False})
+    if not tokenData["id"]:
+        raise InvalidToken
+
+    r = client.list(data[TOKEN])
```

Comment:
This is blocking IO in the event loop, please refactor it to be either async or run in an executor job